query/src/test/java/io/keen/client/java/KeenQueryTest.java

```diff
@@ -0,0 +1,583 @@
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        request.body.writeTo(outputStream);
+        String resultString = outputStream.toString(ENCODING);
+        assertEquals(resultString, TEST_REQ_EVENT_COLLECTION_AND_TARGET_PROP);
+    }
+
+    @Test
+    public void testPercentile()  throws Exception {
+        setMockResponse(200, "{\"result\": 3}");
+        ArgumentCaptor<Request> capturedRequest = ArgumentCaptor.forClass(Request.class);
+        Double result = queryClient.percentile(TEST_EVENT_COLLECTION, TEST_TARGET_PROPERTY, 50.0, null);
+
+        verify(mockHttpHandler).execute(capturedRequest.capture());
+        Request request = capturedRequest.getValue();
+        assertTrue(request.url.toString().contains(KeenQueryConstants.PERCENTILE));
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        request.body.writeTo(outputStream);
+        String resultString = outputStream.toString(ENCODING);
+        assertEquals(resultString, "{\""+KeenQueryConstants.TARGET_PROPERTY+"\":\""+TEST_TARGET_PROPERTY+"\",\"percentile\":50.0,\""+KeenQueryConstants.EVENT_COLLECTION+"\":\""+TEST_EVENT_COLLECTION+"\"}");
+    }
+    @Test
+    public void testSum()  throws Exception {
+        setMockResponse(200, "{\"result\": 65}");
+        ArgumentCaptor<Request> capturedRequest = ArgumentCaptor.forClass(Request.class);
+        Double result = queryClient.sum(TEST_EVENT_COLLECTION, TEST_TARGET_PROPERTY, null);
+
+        verify(mockHttpHandler).execute(capturedRequest.capture());
+        Request request = capturedRequest.getValue();
+        assertTrue(request.url.toString().contains(KeenQueryConstants.SUM_RESOURCE));
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        request.body.writeTo(outputStream);
+        String resultString = outputStream.toString(ENCODING);
+        assertEquals(resultString, TEST_REQ_EVENT_COLLECTION_AND_TARGET_PROP);
+
+    }
+    @Test
+    public void testSelectUnique()  throws Exception {
+        setMockResponse(200, "{\"result\": [0, 1, 2, 3, 4, 5, 6, 7, 8]}");
+        ArgumentCaptor<Request> capturedRequest = ArgumentCaptor.forClass(Request.class);
+        QueryResult result = queryClient.selectUnique(TEST_EVENT_COLLECTION, TEST_TARGET_PROPERTY, null);
+
+        verify(mockHttpHandler).execute(capturedRequest.capture());
+        Request request = capturedRequest.getValue();
+        assertTrue(request.url.toString().contains(KeenQueryConstants.SELECT_UNIQUE_RESOURCE));
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        request.body.writeTo(outputStream);
+        String resultString = outputStream.toString(ENCODING);
+        assertEquals(resultString, TEST_REQ_EVENT_COLLECTION_AND_TARGET_PROP);
+
+        assertTrue(result.isListResult());
+        assertTrue(result.getListResults().get(0).isLong());
+    }
+//
```

Comment:
Nit: Delete these since they're unused. We'll have them in the git history if/when we get around to implementing these features.